src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts

```diff
@@ -705,4 +705,16 @@ export class RgwUserFormComponent implements OnInit {
     result = _.uniq(result);
     return result;
   }
+
+  onMaxBucketsModeChange(mode: string) {
+    if (mode === '1') {
+      // If 'Custom' mode is selected, then ensure that the form field
+      // 'Max. buckets' contains a valid value. Set it to default if
+      // necessary.
+      const maxBuckets = this.userForm.getValue('max_buckets');
+      this.userForm.patchValue({
+        max_buckets: _.isEmpty(_.toString(maxBuckets)) ? 1000 : maxBuckets
+      });
```

Comment:
Currently implementation also fails in some situations.
If you put `-1000` in the input, it will still be `-1000` when you change back to custom.
How about:
```suggestion
      const maxBucketsValid = this.userForm.get('max_buckets').valid;
      if (!maxBucketsValid) {
        this.userForm.patchValue({
          max_buckets: 1000
        });
      }
```

Reply:
Good idea, this way we can decide based on the validators result.